### PR TITLE
Refactor: Splash dan Authentication dengan Unit Test

### PR DESCRIPTION
**Pull request by rfaturriza**
Fitur:

- Splash
- Onboarding
- Auth - Login Email
- Auth - Login No HP
- Auth - Registrasi Email
- Auth - Registrasi No HP
- Auth - Reset Password
- Local Auth - PIN

### CODE REVIEW

**Comment by rfaturriza** [comment]
manual testing:
- fvm flutter pub run  build_runner build --delete-conflicting-outputs
- fvm flutter test

File: test/features/authentication/data/repositories/auth_repository_test.dart [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+      )).thenThrow(FirebaseAuthException(code: 'invalid-email'));
+
+      expect(
+        () => authRepository.forgetEmailPassword(email: email),
+        throwsA(isA<FirebaseAuthException>()),
+      );
+    });
+  });
+
+  // group('verifyPhoneNumber', () {
```
Comment (peltops) [inline test/features/authentication/data/repositories/auth_repository_test.dart]: @rfaturriza what would happen if test case group `verifyPhoneNumber` is enabled?

Comment (peltops) [reply test/features/authentication/data/repositories/auth_repository_test.dart]: @rfaturriza ignore my comment above, I just see the `verifyPhoneNumber` test group is enabled in the last commit

Comment (rfaturriza) [reply test/features/authentication/data/repositories/auth_repository_test.dart]: It's outdated file. You can check the new commit here https://github.com/peltops/e-imunisasi/pull/100/commits/afdca06b4f2d47e83a319887478eb791ff4eb9fe

**Comment by peltops** [comment]
@rfaturriza , is this PR ready for review?

**Comment by faisaladnanpeltops** [comment]
approved, well done!

**Pull request merged**